btree.go

```diff
@@ -347,24 +347,23 @@ func (n *node) growChildAndRemove(i int, item Item, minItems int, typ toRemove)
 	return n.remove(item, minItems, typ)
 }
 
-// iterate provides a simple method for iterating over elements in the tree.
-// It could probably use some work to be extra-efficient (it calls from() a
-// little more than it should), but it works pretty well for now.
+// iterateRange provides a simple method for iterating over all elements in the
+// tree that are within a given range.
 //
 // It requires that 'from' and 'to' both return true for values we should hit
 // with the iterator.  It should also be the case that 'from' returns true for
 // values less than or equal to values 'to' returns true for, and 'to'
 // returns true for values greater than or equal to those that 'from'
 // does.
-func (n *node) iterate(from, to func(Item) bool, iter ItemIterator) bool {
+func (n *node) iterateRange(from, to Item, iter ItemIterator) bool {
 	for i, item := range n.items {
-		if !from(item) {
+		if item.Less(from) {
```

Comment:
use n.items.find here to speed this up a bit?  Now that you're looking for an item, no reason we can't binary-search for it.